engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/ROverseer.java

```diff
@@ -0,0 +1,153 @@
+package com.odysseusinc.arachne.executionengine.execution.r;
+
+import com.odysseusinc.arachne.execution_engine_common.api.v1.dto.ExecutionOutcome;
+import com.odysseusinc.arachne.execution_engine_common.api.v1.dto.Stage;
+import com.odysseusinc.arachne.executionengine.execution.Overseer;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.Instant;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ROverseer implements Overseer {
+    private final ScheduledExecutorService executor = new ScheduledThreadPoolExecutor(1) {{
+        setRemoveOnCancelPolicy(true);
+    }};
+
+    private final long id;
+    private final Process process;
+    private final BufferedReader reader;
+    private final StringBuffer stdout = new StringBuffer();
+    private final BiConsumer<String, String> callback;
+    private final ScheduledFuture<?> logFlush;
+    @Getter
+    private final CompletableFuture<ExecutionOutcome> result;
+    @Getter
+    private final Instant started;
+
+    /**
+     * Creates a new process overseer.
+     *
+     * @param id             Execution identifier, for logging.
+     * @param process        Process to manage
+     * @param timeout        Timeout (in seconds). Once this amount of time is passed, the process will be terminated.
+     * @param callback       Consumer to send progress. First argument is current stage, second is log.
+     * @param updateInterval Log polling interval, in milliseconds.
+     * @param started        The moment when execution has been requested
+     */
+    public ROverseer(long id, Process process, int timeout, BiConsumer<String, String> callback, int updateInterval, Instant started) {
+        this.id = id;
+        this.process = process;
+        reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+        this.callback = callback;
+        this.started = started;
+
+        result = new CompletableFuture<>();
+
+        ScheduledFuture<?> watchdog = executor.schedule(() -> {
```

Comment:
The watchdog variable is unused. Moreover, what this scheduled task is doing?

Reply:
The variable is indeed unnecessary and is removed.
The scheduled task will terminate the execution after `runtime.timeOutSec` seconds.
This is not a new behaviour, however with this PR it is improved to end up with a real termination.

Reply:
Well, then 1 hour is too few. There are analyses that can run about 10-12 hours.